### ISSUE TITLE
sort by columns with CASE_INSENSITIVE

### ISSUE THREAD

**Issue by xperad** [realm/realm-java#2291]
Is there a way to sort by columns with case sensitive or case insensitive?

ex)
```
realm.where(MyTable.class)
    .equalTo("column1", "value1")
    .findAllSorted("column2", Sort.ASCENDING, CASE.SENSITIVE);
```

if no, I hope this function with non-ASCII strongly :)

**Comment by beeender** [comment]
Unfortunately sorting with case insensitive is not supported right now :(

**Comment by bkromhout** [comment]
@beeender Is there any good workaround you could suggest for this? i.e., some way to make the values all upper-/lower-case before the sort happens (but not persist that change to the actual objects)?

My only idea was storing an extra copy of such fields that I'd converted to all lower-case, but that seemed like overkill just to provide case-insensitive sorting, so I figured I'd ask.

**Comment by beeender** [comment]
> My only idea was storing an extra copy of such fields that I'd converted to all lower-case, but that seemed like overkill just to provide case-insensitive sorting, so I figured I'd ask.
@bkromhout Yeah, this is what come to my mind as well... I don't have other better workaround right now...

**Comment by bkromhout** [comment]
That's fine, I've implemented it that way and it works, just have to remember to update that extra sort field if I update the real one.

**Comment by mommyme** [comment]
Have there been any updates to this issue?
